perfkitbenchmarker/packages/ycsb.py

```diff
@@ -47,3 +143,533 @@ def YumInstall(vm):
+  for percentile in percentiles:
+    if percentile < 0 or percentile > 100:
+      raise ValueError('Invalid percentile: {0}'.format(percentile))
+    if math.modf(percentile)[0] < 1e-7:
+      percentile = int(percentile)
+    label = 'p{0}'.format(percentile)
+    latencies, freqs = zip(*histogram)
+    time_ms = _WeightedQuantile(latencies, freqs, percentile * 0.01)
+    result[label] = time_ms
+  return result
+
+
+def _CombineResults(result_list, combine_histograms=True):
+  """Combine results from multiple YCSB clients."""
+  result = copy.deepcopy(result_list[0])
+
+  # Binary operators to aggregate reported statistics.
+  # 'None' will be dropped.
+  operators = {
+      'Operations': operator.add,
+      'RunTime(ms)': max,
+      'Return=0': operator.add,
+      'Return=-1': operator.add,
+      'Return=-2': operator.add,
+      'Return=-3': operator.add,
+      'AverageLatency(ms)': None,
+      'Throughput(ops/sec)': operator.add,
+      '95thPercentileLatency(ms)': None,
+      '99thPercentileLatency(ms)': None,
+      'MinLatency(ms)': min,
+      'MaxLatency(ms)': max}
+
+  def CombineHistograms(hist1, hist2):
+    h1 = dict(hist1)
+    h2 = dict(hist2)
+    keys = sorted(frozenset(h1) | frozenset(h2))
+    result = []
+    for k in keys:
+      result.append((k, h1.get(k, 0) + h2.get(k, 0)))
+    return result
+
+  for indiv in result_list[1:]:
+    for group_name, group in indiv['groups'].iteritems():
+      for k, v in group['statistics'].iteritems():
+        if k not in result['groups'][group_name]['statistics']:
+          logging.warn('"%s" in new dict, but not in old.', k)
+          result['groups'][group_name]['statistics'][k] = copy.deepcopy(v)
+          continue
+        elif k not in operators:
+          logging.warn('No operator for "%s". Skipping aggregation.', k)
+          continue
+        elif operators[k] is None:
+          result['groups'][group_name]['statistics'].pop(k, None)
+          continue
+        result['groups'][group_name]['statistics'][k] = (
+            operators[k](result['groups'][group_name]['statistics'][k], v))
+
+      if combine_histograms:
+        result['groups'][group_name]['histogram'] = CombineHistograms(
+            result['groups'][group_name]['histogram'],
+            group['histogram'])
+      else:
+        result['groups'][group_name].pop('histogram', None)
+    result['client'] = ' '.join((result['client'], indiv['client']))
+    result['command_line'] = ';'.join((result['command_line'],
+                                       indiv['command_line']))
+    if 'target' in result and 'target' in indiv:
+      result['target'] += indiv['target']
+
+  return result
+
+
+def _ParseWorkload(contents):
+  """Parse a YCSB workload file.
+
+  Args:
+    contents: str. Contents of the file.
+
+  Returns:
+    dict mapping from property key to property value for each property found in
+    'contents'.
+  """
+  matches = re.findall(r'^([^#]+?)\s*=\s*(.*)$', contents, re.MULTILINE)
```

Comment:
As best I can tell, [no](http://en.wikipedia.org/wiki/.properties). Agreed that this is fragile though. I've made it a little cleaner, and noted the limitations in the docstring.